unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetBundlesLoader.cs

```diff
@@ -0,0 +1,220 @@
+        private const int SKIPPED_FRAMES_AFTER_BUDGET_TIME_IS_REACHED_FOR_DISTANT_ASSETS = 10;
+        private const float MAX_SQR_DISTANCE_FOR_QUICK_LOADING = 6000f;
+        private const float TIME_BETWEEN_REPRIORITIZATIONS = 1f;
+
+        private struct AssetBundleInfo
+        {
+            public Asset_AB asset;
+            public AssetBundle assetBundle;
+            public Transform containerTransform;
+            public Action onSuccess;
+
+            public AssetBundleInfo(Asset_AB asset, AssetBundle assetBundle, Transform containerTransform, Action onSuccess)
+            {
+                this.asset = asset;
+                this.assetBundle = assetBundle;
+                this.containerTransform = containerTransform;
+                this.onSuccess = onSuccess;
+            }
+        }
+
+        private Coroutine assetBundlesLoadingCoroutine;
+        private IOrderedEnumerable<string> assetsToLoad;
+        private Queue<AssetBundleInfo> assetBundlesReadyToBeLoaded = new Queue<AssetBundleInfo>();
+        private Queue<AssetBundleInfo> assetBundlesWaitingForLoad = new Queue<AssetBundleInfo>();
+        private Dictionary<string, int> loadOrderByExtension = new Dictionary<string, int>()
+        {
+            {"png", 0},
+            {"jpg", 1},
+            {"peg", 2},
+            {"bmp", 3},
+            {"psd", 4},
+            {"iff", 5},
+            {"mat", 6},
+            {"nim", 7},
+            {"ltf", 8},
+            {"glb", 9}
+        };
+        private List<UnityEngine.Object> loadedAssetsByName = new List<UnityEngine.Object>();
+        private float currentLoadBudgetTime = 0;
+        private AssetBundleInfo assetBundleInfoToLoad;
+        private float lastQueuesReprioritizationTime = 0;
+
+        private bool limitTimeBudget => CommonScriptableObjects.rendererState.Get();
+
+        public void Start()
+        {
+            if (assetBundlesLoadingCoroutine != null)
+                return;
+
+            assetBundlesLoadingCoroutine = CoroutineStarter.Start(LoadAssetBundlesCoroutine());
+        }
+
+        public void Stop()
+        {
+            if (assetBundlesLoadingCoroutine == null)
+                return;
+
+            CoroutineStarter.Stop(assetBundlesLoadingCoroutine);
+            assetBundlesReadyToBeLoaded.Clear();
+            assetBundlesWaitingForLoad.Clear();
+            assetsToLoad.ToList().Clear();
+        }
+
+        public void MarkAssetBundleForLoad(Asset_AB asset, AssetBundle assetBundle, Transform containerTransform, Action onSuccess)
+        {
+            CheckForReprioritizeAwaitingAssets();
+
+            AssetBundleInfo assetBundleToLoad = new AssetBundleInfo(asset, assetBundle, containerTransform, onSuccess);
+
+            float distanceFromPlayer = GetDistanceFromPlayer(containerTransform);
+            if (distanceFromPlayer <= MAX_SQR_DISTANCE_FOR_QUICK_LOADING)
+                assetBundlesReadyToBeLoaded.Enqueue(assetBundleToLoad);
```

Comment:
Can we change the queue names to `lowPriorityLoadQueue` and `highPriorityLoadQueue`? This'd look more intuitive to me with those names.

Reply:
Done! :)